boa/src/builtins/set/mod.rs

```diff
@@ -0,0 +1,420 @@
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        let species = WellKnownSymbols::species();
+
+        let species_getter = FunctionBuilder::new(context, Self::species_getter)
+            .callable(true)
+            .constructable(false)
+            .name("get [Symbol.species]")
+            .build();
+
+        let size_getter = FunctionBuilder::new(context, Self::size_getter)
+            .callable(true)
+            .constructable(false)
+            .name("get size")
+            .build();
+
+        let iterator_symbol = WellKnownSymbols::iterator();
+
+        let to_string_tag = WellKnownSymbols::to_string_tag();
+
+        let values_function = FunctionBuilder::new(context, Self::values)
+            .name("values")
+            .length(0)
+            .callable(true)
+            .constructable(false)
+            .build();
+
+        let set_object = ConstructorBuilder::new(context, Self::constructor)
+            .name(Self::NAME)
+            .length(Self::LENGTH)
+            .static_accessor(species, Some(species_getter), None, Attribute::CONFIGURABLE)
+            .method(Self::add, "add", 1)
+            .method(Self::clear, "clear", 0)
+            .method(Self::delete, "delete", 1)
+            .method(Self::entries, "entries", 0)
+            .method(Self::for_each, "forEach", 1)
+            .method(Self::has, "has", 1)
+            .property(
+                "keys",
+                values_function.clone(),
+                Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+            )
+            .accessor("size", Some(size_getter), None, Attribute::CONFIGURABLE)
+            .property(
+                "values",
+                values_function.clone(),
+                Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+            )
+            .property(
+                iterator_symbol,
+                values_function,
+                Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+            )
+            .property(to_string_tag, "Set", Attribute::CONFIGURABLE)
+            .build();
+
+        (Self::NAME, set_object.into(), Self::attribute())
+    }
+}
+
+impl Set {
+    pub(crate) const LENGTH: usize = 0;
+
+    /// Create a new set
+    pub(crate) fn constructor(
+        new_target: &Value,
+        args: &[Value],
+        context: &mut Context,
+    ) -> Result<Value> {
+        // 1
+        if new_target.is_undefined() {
+            return context
+                .throw_type_error("calling a builtin Set constructor without new is forbidden");
+        }
+
+        // 2
+        let set_prototype = context
+            .global_object()
+            .get(&"Set".into(), context.global_object().into(), context)?
+            .get_field(PROTOTYPE, context)?
+            .as_object()
+            .expect("'Set' global property should be an object");
```

Comment:
we should store the constructor and prototype in the [`StandardOobjects`](https://github.com/boa-dev/boa/blob/master/boa/src/context.rs#L80-L99).

Reply:
then we only need to do:
```rust
let prototype = context.standard_objects().set_object().prototype();
```

Reply:
Can you explain a bit more why we want this? Is this so that people using boa can get these StandardObjects quickly from the Context instead of doing a lookup in a HashMap?